src/auth/digest/Config.cc

```diff
@@ -826,11 +826,14 @@ Auth::Digest::Config::decode(char const *proxy_auth, const HttpRequest *request,
             break;
 
         case DIGEST_NC:
-            if (value.size() != 8) {
+            if (value.size() == 8) {
+                // For historical reasons, the nc value MUST be exactly 8 hexadecimal
+                // digits.
```

Comment:
Since this code does not validate the contents of `nc`, I would relax the comment to avoid implying otherwise:

```suggestion
                // for historical reasons, the nc value MUST be exactly 8 bytes
```